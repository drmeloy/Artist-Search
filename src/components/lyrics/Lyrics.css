
@import url('https://fonts.googleapis.com/css?family=Raleway:300&display=swap');
@import url('https://fonts.googleapis.com/css?family=Raleway:800&display=swap');
@import url('https://fonts.googleapis.com/css?family=Montserrat:300&display=swap');

.Lyrics {
  position: fixed;
  top: 0;
  width: 65vw;
  height: 100vh;
  padding-left: 60px;
  display: flex;
  flex-direction: column;
  background: black;
  animation: slidein .5s 1;
  overflow: auto;

  /* Any body text gets Montserrat, font-weight 300 */
  p {
    width: 70%;
    line-height: 1.5;
    font-family: 'Montserrat', sans-serif;
    font-weight: 300;
    color: white;
    white-space: pre-line;
  }

  /* H1 and Headers get Raleway Heavy, font-weight: 800 */
  h1 {
    font-family: 'Raleway', sans-serif;
    font-weight: 800;
    font-size: 3rem;
    color: white;
  }

  /* Sub-Headings get Raleway, font-weight 300 */
  h2 {
    font-family: 'Raleway', sans-serif;
    font-weight: 300;
    font-size: 3rem;
    color: white;
  }

  button {
    position: fixed;
    right: 35vw;
    bottom: 2vh;
    font-size: 1rem;
    color: white;
    background: #000000;
  }

  @keyframes slidein {
    from {
      left: -100vw;
    }

    to {
      left: 0;
    }
  }
}

.SlideOut {
  animation: slideout .5s 1;
  
  @keyframes slideout {
    from {
      left: 0;
    }

    to {
      left: -100vw;
    }
  }
}