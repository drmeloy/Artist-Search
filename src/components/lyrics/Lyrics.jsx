import React from 'react';
import PropTypes from 'prop-types';
import { useGetLyrics } from '../hooks/lyrics';
import styles from './Lyrics.css';
import { useHistory } from 'react-router-dom';

const Lyrics = ({ match }) => {
  const history = useHistory();
  const { artistName, artistId, albumId, songTitle } = match.params;

  const goBack = ({ target }) => {
    target.parentNode.classList.add(styles.SlideOut);
    setTimeout(() => {
      history.replace(`/${artistName}/${artistId}/${albumId}`);
    }, 500);
  };

  const lyrics = useGetLyrics(artistName, songTitle);

  return (
    <section className={styles.Lyrics}>
      <h1>{songTitle}</h1>
      <h2>By {artistName}</h2>
      <p>{lyrics}</p>
      <button onClick={goBack}>&#8672;</button>
    </section>
  );
};

Lyrics.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      artistName: PropTypes.string.isRequired,
      songTitle: PropTypes.string.isRequired,
      artistId: PropTypes.string.isRequired,
      albumId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Lyrics;
