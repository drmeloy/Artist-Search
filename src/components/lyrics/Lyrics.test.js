import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from './Lyrics';

describe('Lyrics component', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<Lyrics match={{ params: { artistName: '', songTitle: '' }}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
