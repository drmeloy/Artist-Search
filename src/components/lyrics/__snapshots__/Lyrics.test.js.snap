// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics component matches its snapshot 1`] = `
<Fragment>
  <h1 />
  <h2>
    By 
  </h2>
  <p />
</Fragment>
`;
